PSI_Tracker.js

```diff
@@ -0,0 +1,645 @@
+ */
+function callPSIAPIWithScreenshots() {
+  performTasks("PSI", true)
+}
+
+/**
+ * Call performTasks just for the CrUX API
+ */
+function callCrUX() {
+  performTasks("CrUX")
+}
+
+/**
+ * Call performTasks for CrUX History
+ */
+function callCrUXHistory() {
+  performTasks("CrUX History")
+}
+
+/**
+ * Call performTasks for Green Domain Dataset
+ */
+function callGreenDomain() {
+  performTasks("Green Domain")
+}
+
+/**
+ * Call performTasks for Accessibility
+ */
+function callAccessibility() {
+  performTasks("Accessibility")
+}
+
+/**
+ * Call performTasks for Sustainability
+ */
+function callSustainability() {
+  performTasks("Sustainability")
+}
+
+
+
+/**
+ * Simple runBatch function to be called by trigger
+ */
+function runBatchPSI() {
+  runBatch("PSI")
+}
+
+
+/**
+ * Simple runBatch function to be called by trigger
+ */
+function runBatchPSIWithScreenshots() {
+  runBatch("PSI", true)
+}
+
+
+/**
+ * Simple runBatch function to be called by trigger
+ */
+function runBatchCrUXHistory() {
+  runBatch("CrUX History")
+}
+
+
+/**
+ * Simple runBatch function to be called by trigger
+ */
+function runBatchCrUX() {
+  runBatch("CrUX")
+}
+
+
+/**
+ * Simple runBatch function to be called by trigger
+ */
+function runBatchGreenDomain() {
+  runBatch("Green Domain")
+}
+
+/**
+ * Simple runBatch function to be called by trigger
+ */
+function runBatchAccessibility() {
+  runBatch("Accessibility")
+}
+
+
+/**
+ * Simple runBatch function to be called by trigger
+ */
+function runBatchSustainability() {
+  runBatch("Sustainability")
+}
+
+
+
+
+/**
+ * Parse URLs from the "Config" tab and sends it to the selected API
+ */
+function performTasks(api = "PSI", save_screenshot = false) {
+
+    // Show a warning that the data will be send to a 3P that might collect information
+    if(api == "Green Domain"){
+      if(showAlertCallGWF()){
+        // if permission declined, abort.
+        return true;
+      }
+    }
+
+    // Show a warning that the data will be send to a 3P that might collect information
+    if(api == "Sustainability"){
+      if(showAlertSustainability()){
+        // if permission declined, abort.
+        return true;
+      }
+    }
+
+    // Add the loading symbols and initialise a blank state
+    initURLStatus();
+    // Run batches
+    const execution_mode_boolean = CONFIG_SHEET.getRange("B24").getValue();
+    const execution_mode = execution_mode_boolean ? "TRIGGER" : "SERIAL";
+    // Default execution mode: queries run one after the other and might timoute
+    if(execution_mode == "SERIAL"){
+      while(runBatch(api, save_screenshot)){
+          Logger.log("Batch done")
+      }
+      toast("Done","Finished")
+    }
+    // Advanced mode with triggers
+    else if(execution_mode == "TRIGGER"){
+      removeBatchTriggers();
+      SpreadsheetApp.flush();
+      const TIME_BETWEEN_BATCHES = CONFIG_SHEET.getRange("B27").getValue();
+      if(api == "PSI" && save_screenshot == true){
```

Comment:
might be clearer as a switch?

Reply:
I thought switch was only on one variable name, here I want to have multiple scenarios for both `api` and `save_screenshot`, I could do a switch on `api` and a second test on `save_screenshot`, wdyt?